admin/cluster_containers_setup.py

```diff
@@ -0,0 +1,324 @@
+                '--cert-directory <path where all the certificates are>'
+                '[--cluster <file> (UNUSED)')
+
+    def postOptions(self):
+        if self['app-template'] is not None:
+            template_file = FilePath(self['app-template'])
+            self['template'] = yaml.safe_load(template_file.getContent())
+        elif self['apps-per-node'] > 0:
+            raise usage.UsageError(
+                "app-template parameter must be provided if apps-per-node > 0"
+            )
+
+        if self['wait'] is not None:
+            try:
+                self['wait'] = int(self['wait'])
+            except ValueError:
+                raise usage.UsageError("The wait timeout must be an integer.")
+
+
+def main(reactor, argv):
+    environ = os.environ
+    try:
+        options = ContainerOptions()
+        options.parseOptions(argv[1:])
+    except usage.UsageError as e:
+        sys.stderr.write(e.args[0])
+        sys.stderr.write('\n\n')
+        sys.stderr.write(options.getSynopsis())
+        sys.stderr.write('\n')
+        sys.stderr.write(options.getUsage())
+        raise SystemExit(1)
+    container_deployment = ClusterContainerDeployment(reactor,
+                                                      environ,
+                                                      options)
+    return container_deployment.deploy_and_wait_for_creation()
+
+
+class ResponseError(Exception):
+    """
+    An unexpected response from the REST API.
+    """
+    def __init__(self, code, message):
+        Exception.__init__(self, "Unexpected response code {}:\n{}\n".format(
+            code, message))
+        self.code = code
+
+
+class ClusterContainerDeployment(object):
+    """
+    Class that contains all the methods needed to deploy a new config in a
+    cluster.
+
+    :ivar options: ``ContainerOptions`` with the options passed to the script
+    :ivar application_template: template of the containers to deploy.
+    :ivar per_node: number of containers and dataset per node.
+    :ivar control_node_address: public ip address of the control node.
+    :ivar cluster_cert: ``FilePath`` of the cluster certificate.
+    :ivar user_cert: ``FilePath`` of the user certificate.
+    :ivar user_key: ``FilePath`` of the user key.
+    :ivar client: ``FlockerClient`` conected to the cluster.
+    :ivar reactor: ``Reactor`` used by the client.
+    :ivar nodes: list of `Node` returned by client.list_nodes
+    """
+    def __init__(self, reactor, env, options):
+        """
+        ``ClusterContainerDeployment`` constructor.
+
+        :param reactor: ``Reactor`` we are using.
+        :param env: ``environ`` with the current environment.
+            NOTE: alternative of making it work with env variables pending
+            implementation
+        :param options: ``ContainerOptions`` with the options passed to the
+            the script.
+        """
+        # XXX add the capability to use env variables if the options are not
```

Comment:
:question: A fairly major change that would help here is to put this code in the `benchmark` directory (the only place it is likely to be used) and use the `BenchmarkCluster` class, which accepts either the env vars or the UFT installer configuration, and provides the control server object.